MetovaTestKit/UIKitTesting/UIAlertControllerAssertions.swift

```diff
@@ -0,0 +1,146 @@
+///   - line: The line number to report the failure for. The default value will report the line number of the calling site.
+public func MTKAssertAlertIsPresented(by presenter: UIViewController, style: UIAlertControllerStyle, title: String?, message: String?, actions: [ExpectedAlertAction], _ failureMessage: String? = nil, file: StaticString = #file, line: UInt = #line) {
+    
+    guard let alert = presenter.presentedViewController as? UIAlertController else {
+        MTKRecordFailure(withMessage: failureMessage, description: "No alert was presented.", file: file, line: line)
+        return
+    }
+    
+    guard alert.preferredStyle == style else {
+        MTKRecordFailure(withMessage: failureMessage, description: "Expected alert to have style `\(style)`. Instead found `\(alert.preferredStyle)`.", file: file, line: line)
+        return
+    }
+    
+    guard alert.title == title else {
+        MTKRecordFailure(withMessage: failureMessage, description: "Expected alert to have title \(quotedString(title)). Instead found \(quotedString(alert.title)).", file: file, line: line)
+        return
+    }
+    
+    guard alert.message == message else {
+        MTKRecordFailure(withMessage: failureMessage, description: "Expected alert to have message \(quotedString(message)). Instead found \(quotedString(alert.message)).", file: file, line: line)
+        return
+    }
+    
+    let doActionsMatchExpectedActions: Bool = {
+        
+        guard actions.count == alert.actions.count else { return false }
+        
+        for (expectedAction, actualAction) in zip(actions, alert.actions) where !expectedAction.matchesActualAction(actualAction) {
+            return false
+        }
+        
+        return true
+    }()
+    
+    guard doActionsMatchExpectedActions else {
+        
+        let expectedActionDescriptions = actions.map({ "(title: \"\($0.title)\", style: \($0.style))" }).commaSeparatedList()
+        
+        let actualActionDescriptions = alert.actions.map({ "(title: \(quotedString($0.title)), style: \($0.style))" }).commaSeparatedList()
+        
+        MTKRecordFailure(withMessage: failureMessage, description: "Expected alert to have actions: \(expectedActionDescriptions). Instead found \(actualActionDescriptions).", file: file, line: line)
+        
+        return
+    }
+}
+
+// MARK: - ExpectedAlertAction
+
+/// A type used to represent an expected title and style for a `UIAlertAction` instance.
+public struct ExpectedAlertAction {
+    
+    // MARK: Properties
+    
+    fileprivate let title: String
+    fileprivate let style: UIAlertActionStyle
+    
+    // MARK: Initialization
+    
+    public init(title: String, style: UIAlertActionStyle) {
+        
+        self.title = title
+        self.style = style
+    }
+    
+    // MARK: Action Comparison
+
+    fileprivate func matchesActualAction(_ action: UIAlertAction) -> Bool {
```

Comment:
Couldn't we just implement a `==` function for `UIAlertAction` (and make it `Equatable`)?  If we did that, we could just use `XCTAssertEqual` on them.